ext/rugged/rugged_tree.c

```diff
@@ -614,15 +614,19 @@ void rugged_parse_merge_options(git_merge_options *opts, VALUE rb_options)
 		}
 
 		if (RTEST(rb_hash_aref(rb_options, CSTR2SYM("renames")))) {
-			opts->tree_flags |= GIT_MERGE_TREE_FIND_RENAMES;
+			opts->flags |= GIT_MERGE_FIND_RENAMES;
 		}
 
 		if (RTEST(rb_hash_aref(rb_options, CSTR2SYM("fail_on_conflict")))) {
-			opts->tree_flags |= GIT_MERGE_TREE_FAIL_ON_CONFLICT;
+			opts->flags |= GIT_MERGE_FAIL_ON_CONFLICT;
 		}
 
 		if (RTEST(rb_hash_aref(rb_options, CSTR2SYM("skip_reuc")))) {
-			opts->tree_flags |= GIT_MERGE_TREE_SKIP_REUC;
+			opts->flags |= GIT_MERGE_SKIP_REUC;
+		}
+
+		if (RTEST(rb_hash_aref(rb_options, CSTR2SYM("no_recursive")))) {
+			opts->flags |= GIT_MERGE_NO_RECURSIVE;
```

Comment:
`no_recursive` makes sense as a flag. We'll leave it like this and add the flag by default to the options in GitRPC.